pkg/namespacestore/reconciler.go

```diff
@@ -771,5 +833,55 @@ func (r *Reconciler) ReconcileNamespaceStore() error {
 			return err
 		}
 	}
+
+	if r.CreateNamespaceResourceParams.NSFSConfig != nil {
+		pvcName := r.NamespaceStore.Spec.NSFS.PvcName
+		if !util.KubeCheck(r.DeploymentEndpoint) {
+			r.Logger.Infof("No Endpoint deployment yet. requeuing for 5 minutes")
+			return fmt.Errorf("No Endpoint Deployment. requeue again")
+		}
+		isPvcExist := false
+		volumeName := "nsfs" + util.RandomHex(4)
+		for _, volume := range r.DeploymentEndpoint.Spec.Template.Spec.Volumes {
+			if volume.PersistentVolumeClaim != nil && volume.PersistentVolumeClaim.ClaimName == pvcName {
+					isPvcExist = true // PVC already attached to the pods - no need to add
+					volumeName = volume.Name
+					break;
+			}
+		}
+		if (!isPvcExist) {
+			r.DeploymentEndpoint.Spec.Template.Spec.Volumes = append(r.DeploymentEndpoint.Spec.Template.Spec.Volumes, corev1.Volume {
+				Name: volumeName,
+				VolumeSource: corev1.VolumeSource {
+					PersistentVolumeClaim: &corev1.PersistentVolumeClaimVolumeSource{
+						ClaimName: pvcName,
+					},
+				},
+			})
+		}
+		isMountExist := false
```

Comment:
might be Esyer to call it `willAddMount` and switch the logic
WDYT?

Reply:
I kind of prefer it this way - this what I'm checking for

Reply:
changed all logic - not relevant